src/Stratis.Features.FederatedPeg/Wallet/FederationWalletManager.cs

```diff
@@ -473,6 +472,8 @@ public bool RemoveTransaction(Transaction transaction)
                 TransactionData spentTransaction = this.Wallet.MultiSigAddress.Transactions.SingleOrDefault(t => (t.Id == tTx.Id) && (t.Index == tTx.Index));
                 if (spentTransaction != null)
                 {
+                    this.logger.LogTrace("Unspending {0}-{1}", spentTransaction.Id, spentTransaction.Index);
```

Comment:
I'm prettttty sure we're hitting this when we shouldn't be. And it's allowing UTXOs to be spent again.

This will help us confirm.